### PR TITLE
Implement several fixes as suggested in the last facilitator application bug bash

### PR DESCRIPTION
**Pull request by aoby**
1. Add status and notes to CSV download
 2. Remove csf-specific fields for csp/csd applications and vice versa
 3. Clear irrelevant answers, meaning for questions that disappear based on other answers for each page on the client before validation.
 4. Removed some stylized quotes (replaces with regular ones) in the label strings. I noticed these were showing up as strange symbols in the CSV download.
 5. Fixed several label keys that were in the wrong format, and split up some long lines.

### CODE REVIEW

File: apps/src/code-studio/pd/application/facilitator1819/Section3LeadingStudents.jsx [hunk 1]
```diff
@@ -86,4 +86,25 @@ export default class Section3LeadingStudents extends Facilitator1819FormComponen
 
     return requiredFields;
   }
+
+  /**
+   * @override
+   */
+  static processPageData(data) {
```
Comment (mehalshah) [inline apps/src/code-studio/pd/application/facilitator1819/Section3LeadingStudents.jsx:93]: Maybe a future todo would be to improve the form controller to clear hidden fields as needed

File: apps/src/code-studio/pd/form_components/FormController.jsx [hunk 2]
```diff
@@ -338,6 +338,8 @@ export default class FormController extends React.Component {
         pageData[field] = value;
       }
     });
+
+    pageData = {...pageData, ...currentPage.processPageData(pageData)};
```
Comment (mehalshah) [inline apps/src/code-studio/pd/form_components/FormController.jsx]: Can this just be done with `Object.assign(pageData, currentPage.processPageData(pageData)` ?

File: lib/cdo/shared_constants/pd/facilitator1819_application_constants.rb [hunk 3]
```diff
@@ -48,16 +48,16 @@ def self.clean_multiline(string)
          For more details about the requirements to facilitate each program, please visit the
          [2018-19 Facilitator Development Program Description](https://docs.google.com/document/d/1aX-KH-t6tgjGk2WyvJ7ik7alH4kFTlZ0s1DsrCRBq6U).'
       ),
-      planOnTeaching: 'Do you plan on teaching this course in the 2018-19 school year?',
-      abilityToMeetRequirements: clean_multiline(
+      plan_on_teaching: 'Do you plan on teaching this course in the 2018-19 school year?',
```
Comment (mehalshah) [inline lib/cdo/shared_constants/pd/facilitator1819_application_constants.rb:51]: This _shouldn't_ break the question numbering, but it's worth checking on your machine to make sure. Please remerge staging to verify

File: apps/src/code-studio/pd/form_components/FormController.jsx (hunk 2 above)
Comment (aoby) [reply apps/src/code-studio/pd/form_components/FormController.jsx]: yeah that should work. 👍

File: apps/src/code-studio/pd/application/facilitator1819/Section3LeadingStudents.jsx (hunk 1 above)
Comment (aoby) [reply apps/src/code-studio/pd/application/facilitator1819/Section3LeadingStudents.jsx:93]: Yeah that would be nice. As it stands, we don't have a good way to discover which components are visible short of inspecting the DOM (or React component tree I suppose) for those names. However, we could improve the way we render conditional components in such a way that is more imperative, maybe with a container component or function, that will take care of the dynamic required fields and data processing. If I have some time after the initial Teacher Application implementation I'll take a look at streamlining this.

**Comment by aoby** [comment]
Circle failure is an unrelated known issue w fallback player. Merging...

**Pull request merged**